utils/upload.go

```diff
@@ -0,0 +1,171 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"zestream-server/constants"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+// channel to extract files from the folder
+type fileWalk chan string
+
+type Uploader interface {
+	Upload(walker fileWalk)
+}
+
+func UploadToCloudStorage(uploader Uploader, path string) {
```

Comment:
UploadToCloudStorage is the function which takes the uploader interface and the path of the folder. First I extracted the files from the folder and then used the Upload method of the interface to upload according to the cloud storage.
For example
To upload folder on aws s3 bucket we first create an instance of AwsUploader and pass the instance along with path to the UploadToCloudStorage function. 
 
	
	```uploader := utils.AwsUploader{ Prefix:   "prefix_name", Session: "session_created"}
	//calling the upload function with uploader interface and the path
	utils.UploadToCloudStorage(&uploader, "C:\\Development\\zestream-server\\testing")```
	
	Makes sense?

Reply:
Yes, Understood thanks for the clarification!!

Reply:
This would make better sense if you look at the screenshot, I did make a demo file to run the functions. 
Do you want me to change this design and create three functions taking paths and the parameters?
Which approach would make adding new cloud storage easier
@uzaxirr